src/entities.rs

```diff
@@ -400,10 +400,13 @@ impl Entities {
     }
 
     pub fn contains(&self, entity: Entity) -> bool {
-        // Note that out-of-range IDs are considered to be "contained" because
-        // they must be reserved IDs that we haven't flushed yet.
         match self.meta.get(entity.id as usize) {
-            Some(meta) => meta.generation == entity.generation,
+            Some(meta) => {
+                meta.generation == entity.generation
+                    && (meta.location.index != u32::MAX
+                        || self.pending[self.free_cursor.load(Ordering::Relaxed).max(0) as usize..]
```

Comment:
This looks heavyweight from a performance perspective. Is this unlikely to be hit?

Reply:
Short-circuiting ensures it only arises on reserved or "phantom" entities. Further, I don't think `contains` is typically hot-loop for people, although that's speculative. We could also drop it and just redefine the semantics of `contains` to exclude reserved but unflushed entities; that seems like it would generally be fine, but on the other hand it'd be difficult to emulate externally if someone does care about that case.

Reply:
> We could also drop it and just redefine the semantics of contains to exclude reserved but unflushed entities;

I think its better to wait if people complain about the performance than the weaken the semantics.